pkg/installer/deployment.go

```diff
@@ -115,52 +101,303 @@ func createDeployment(ctx context.Context, c *Config) error {
 				Ports:          commonContainerPorts,
 				ReadinessProbe: &corev1.Probe{ProbeHandler: readinessHandler},
 				Env:            []corev1.EnvVar{kubeNodeNameEnvVar, csiEndpointEnvVar},
-				VolumeMounts:   volumeMounts,
+				VolumeMounts: []corev1.VolumeMount{
+					newVolumeMount(volumeNameSocketDir, socketDir, corev1.MountPropagationNone, false),
+				},
+			},
+		},
+	}
+
+	selectorValue := fmt.Sprintf("%v-%v", consts.ControllerServerName, getRandSuffix())
+	replicas := int32(3)
+	deployment := &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "apps/v1",
+			Kind:       "Deployment",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        consts.ControllerServerName,
+			Namespace:   consts.Identity,
+			Annotations: map[string]string{},
+			Labels:      defaultLabels,
+			Finalizers:  []string{deploymentFinalizer},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: metav1.AddLabelToSelector(&metav1.LabelSelector{}, selectorKey, selectorValue),
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      consts.ControllerServerName,
+					Namespace: consts.Identity,
+					Annotations: map[string]string{
+						createdByLabel: pluginName,
+					},
+					Labels: map[string]string{
+						selectorKey: selectorValue,
+					},
+				},
+				Spec: podSpec,
+			},
+		},
+		Status: appsv1.DeploymentStatus{},
+	}
+
+	if args.DryRun {
+		fmt.Print(mustGetYAML(deployment))
+		return nil
```

Comment:
In  dry-run mode, we should also print the admin-service

Reply:
It is done at `createAdminService()`

Reply:
when `--dry-run` is set, we return `nil` right away without printing the admin service

Reply:
Fixed `createAdminServerDeployment()`